api/v1/client/src/commonMain/kotlin/OrtServerClient.kt

```diff
@@ -0,0 +1,87 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.client
+
+import io.ktor.client.HttpClient
+import io.ktor.client.plugins.auth.Auth
+import io.ktor.client.plugins.auth.providers.BearerTokens
+import io.ktor.client.plugins.auth.providers.bearer
+
+import kotlinx.serialization.json.Json
+
+import org.eclipse.apoapsis.ortserver.client.api.RepositoriesApi
+import org.eclipse.apoapsis.ortserver.client.auth.AuthService
+
+class OrtServerClient(
+    /**
+     * The configured HTTP client for the interaction with the API.
+     */
+    client: HttpClient,
+
+    /**
+     * The configuration for the ORT server client.
+     */
+    config: OrtServerClientConfig
+) {
+    companion object {
+        /**
+         * Create a new instance of the ORT server client using the given [config] and configure an HTTP client with
+         * the necessary authentication.
+         */
+        fun create(config: OrtServerClientConfig): OrtServerClient {
+            val json = Json {
+                ignoreUnknownKeys = true
+                prettyPrint = true
+                isLenient = true
+            }
+
+            val auth = AuthService(
+                client = createHttpClient(json),
+                tokenUrl = config.tokenUrl,
+                clientId = config.clientId
+            )
+
+            val client = createHttpClient(json) {
+                install(Auth) {
+                    bearer {
+                        loadTokens {
+                            val tokenInfo = auth.generateToken("deo1imb", "password")
+
+                            BearerTokens(tokenInfo.accessToken, tokenInfo.refreshToken)
+                        }
+
+                        refreshTokens {
+                            val tokenInfo = runCatching {
+                                auth.refreshToken(oldTokens?.refreshToken.orEmpty())
+                            }.getOrElse {
+                                auth.generateToken(config.username, config.password)
+                            }
+
+                            BearerTokens(tokenInfo.accessToken, tokenInfo.refreshToken)
+                        }
+                    }
+                }
+            }
+            return OrtServerClient(client, config)
+        }
+    }
+
+    val repositories = RepositoriesApi("${config.baseUrl}/repositories", client)
```

Comment:
IIUC, you expect that the `baseUrl` contains the path with API version. Since the client is specific to a concrete version, it should probably also be responsible for adding the correct version-specific part. So, the `baseUrl` should only contain the host name of the server I think. This should also be documented in the `OrtServerClientConfig` class.